### ISSUE TITLE
Circuit cutting capabilities

### ISSUE THREAD

**Issue by perlinm** [quantumlib/Cirq#6012]
**Is your feature request related to a use case or problem? Please describe.**

There is a growing literature on circuit cutting / circuit knitting / entanglement forging etc.  A few examples: [1](https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.125.150504), [2](https://dl.acm.org/doi/abs/10.1145/3445814.3446758), [3](https://www.nature.com/articles/s41534-021-00390-6), [4](https://arxiv.org/abs/2205.00016), [5](https://arxiv.org/abs/2302.03366), [6](https://journals.aps.org/prxquantum/abstract/10.1103/PRXQuantum.3.010309).  In short, the general idea is to "cut" quantum circuits into smaller subcircuits, run the subcircuits (and minor variants thereof), and classically recombine subcircuit execution results to reconstruct the output of the original circuit.  For circuits with a suitable ("clustered") structure, these techniques reduce overall (classical + quantum) computing costs of estimating the output of a quantum circuit.

It would be nice to have some of these techniques added to Cirq as "experimental" features.

**Describe the solution you'd like**

I propose integrating the codes in [this repository](https://github.com/Quantum-Software-Tools/QSPLIT-MLFT) (from example 3 above), and more specifically [this file](https://github.com/Quantum-Software-Tools/QSPLIT-MLFT) into someplace like `cirq.experiments` (or elsewhere).  I am willing to write an RFC, but before that I would like to gauge the level of potential interest from Cirq developers.

**What is the urgency from your perspective for this issue? Is it blocking important work?**

P3 - I'm not really blocked by it, it is an idea I'd like to discuss / suggestion based on principle
